### ISSUE TITLE
Computation efficiency of in-context generation

### ISSUE THREAD

**Issue by eugenelyj** [River-Zhang/ICEdit#44]
In-context generation looks good, but doubling the number of tokens and will lead to a significant increase in overhead. Are there any good ideas or papers in this area?

**Comment by River-Zhang** [comment]
Hi, thanks for your interest! Actually the tokens we use is similar to the ways of OminiControl, ACE++, UNO and DreamO. Currently our training code enables up to 1024 res image (maybe more) training with gradient checkpointing. And for much bigger resolution it may encounter OOM and maybe you can use deepspeed to reduce VRAM cost.

**Comment by eugenelyj** [comment]
I see. I just want a alternative way that works similar but can be more efficient. The working principle of in-context learning is self-attention between the reference image and the target generation image. Maybe if I create a cross attention layer that init from the self-attn and apply cross-attn from ref to the target, can works similar?
